api/nodes/serializers.py

```diff
@@ -266,6 +269,104 @@ def update(self, node, validated_data):
         return node
 
 
+class NodeAddonSettingsSerializer(JSONAPISerializer):
+    class Meta:
+        type_ = 'node_addons'
+
+    id = ser.CharField(source='_id', read_only=True)
+    enabled = ser.BooleanField(required=True)
+    external_account_id = ser.CharField(source='settings.external_account._id', allow_null=True)
+    folder_id = ser.CharField(source='settings.folder_id', allow_null=True)
+    folder_path = ser.CharField(source='settings.folder_path', required=False, allow_null=True)
+    node_has_auth = ser.BooleanField(source='settings.has_auth', read_only=True)
+    complete = ser.BooleanField(source='settings.complete', read_only=True)
+    configured = ser.BooleanField(source='settings.configured', read_only=True)
+
+    def update(self, instance, validated_data):
+        addon_name = instance.get('_id', None)
+        if addon_name not in ADDONS_FOLDER_CONFIGURABLE:
+            raise exceptions.MethodNotAllowed('Requested addon not currently configurable via API.')
+
+        auth = get_user_auth(self.context['request'])
+        node_settings = instance.get('settings', None)
+
+        try:
+            external_account_id = validated_data['settings']['external_account']['_id']
+            set_auth = True
+        except KeyError:
+            external_account_id = None
+            set_auth = False
+
+        try:
+            folder_id = validated_data['settings']['folder_id']
+            set_folder = True
+        except KeyError:
+            folder_id = None
+            set_folder = False
+
+        if folder_id and addon_name == 'googledrive':
+            try:
+                folder_path = validated_data['settings']['folder_path']
+            except KeyError:
+                folder_path = None
+            folder_id = {
+                'id': folder_id,
+                'path': folder_path
+            }
+
+        enabled = validated_data.get('enabled', None)
+
+        if enabled is False and (folder_id or external_account_id):
+            raise Conflict('Cannot disable the addon and set the folder/authorization at the same time.')
```

Comment:
Given this it could be good to factor these branches into separate helper methods on the class, or at the very least throw in a return after L332.